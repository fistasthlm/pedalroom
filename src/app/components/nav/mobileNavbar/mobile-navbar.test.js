import React from 'react';
import { shallow } from 'enzyme';
import { Map, List } from 'immutable';
import MobileNavbar from './mobile-navbar';

const setup = propOverrides => {
    const props = Object.assign({
        links: List(),
        toggleMenu: jest.fn(),
    }, propOverrides);

    const wrapper = shallow(<MobileNavbar {...props} />);

    return {
        setup,
        wrapper,
    };
};


describe('Render', () => {
    it('should render component', () => {
        const { wrapper } = setup();

        expect(wrapper).toMatchSnapshot();
    });

    it('should render links', () => {
        const { wrapper } = setup({
            links: List([
                Map({
                    to: '/',
                    route: 'Home',
                }),
                Map({
                    to: '/bikes',
                    route: 'Bikes'
                })
            ])
        });

        expect(wrapper).toMatchSnapshot();
    });
});
