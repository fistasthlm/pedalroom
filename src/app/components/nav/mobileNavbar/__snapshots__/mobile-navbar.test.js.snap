// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `<div />`;

exports[`Render should render links 1`] = `
<div>
  <div
    className="menu-item"
    key="0-/"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="nav-item"
      exact={true}
      onClick={[MockFunction]}
      to="/"
    >
      Home
    </NavLink>
  </div>
  <div
    className="menu-item"
    key="1-/bikes"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="nav-item"
      exact={true}
      onClick={[MockFunction]}
      to="/bikes"
    >
      Bikes
    </NavLink>
  </div>
</div>
`;
